src/WebJobs.Script/Workers/SharedMemoryDataTransfer/SharedMemoryManager.cs

```diff
@@ -0,0 +1,313 @@
+            {
+                Name = mapName,
+                Count = contentSize
+            };
+            return response;
+        }
+
+        /// <summary>
+        /// Writes the given <see cref="string"/> data into a <see cref="SharedMemoryMap"/>.
+        /// Note:
+        /// Tracks the reference to the <see cref="SharedMemoryMap"/> after creating it and does not close it.
+        /// </summary>
+        /// <param name="content">Content to write into shared memory.</param>
+        /// <returns>Metadata of the shared memory map into which data is written if successful, <see cref="null"/> otherwise.</returns>
+        private Task<SharedMemoryMetadata> PutStringAsync(string content)
+        {
+            byte[] contentBytes = Encoding.UTF8.GetBytes(content);
+            return PutBytesAsync(contentBytes);
+        }
+
+        /// <summary>
+        /// Reads data as <see cref="byte[]"/> from the <see cref="SharedMemoryMap"/> with the given name.
+        /// </summary>
+        /// <param name="mapName">Name of the <see cref="MemoryMappedFile"/> to read from.</param>
+        /// <param name="offset">Offset to start reading data from in the <see cref="SharedMemoryMap"/>.</param>
+        /// <param name="count">Number of bytes to read from, starting from the offset, in the <see cref="SharedMemoryMap"/>.</param>
+        /// <returns>Data read as <see cref="byte[]"/> if successful, <see cref="null"/> otherwise.
+        /// </returns>
+        private async Task<byte[]> GetBytesAsync(string mapName, int offset, int count)
+        {
+            SharedMemoryMap sharedMemoryMap = Open(mapName);
+
+            try
+            {
+                byte[] content = await sharedMemoryMap.GetBytesAsync(offset, count);
+
+                if (content == null)
+                {
+                    _logger.LogError($"Cannot read content from MemoryMappedFile: {mapName}");
+                    throw new Exception($"Cannot read content from MemoryMappedFile: {mapName} with offset: {offset} and count: {count}");
+                }
+
+                return content;
+            }
+            finally
+            {
+                sharedMemoryMap.Dispose(deleteFile: false);
+            }
+        }
+
+        /// <summary>
+        /// Reads data as <see cref="string"/> from the <see cref="SharedMemoryMap"/> with the given name.
+        /// </summary>
+        /// <param name="mapName">Name of the <see cref="MemoryMappedFile"/> to read from.</param>
+        /// <param name="offset">Offset to start reading data from in the <see cref="SharedMemoryMap"/>.</param>
+        /// <param name="count">Number of bytes to read from, starting from the offset, in the <see cref="SharedMemoryMap"/>.</param>
+        /// <returns>Data read as <see cref="string"/> if successful, <see cref="null"/> otherwise.
+        /// </returns>
+        private async Task<string> GetStringAsync(string mapName, int offset, int count)
```

Comment:
Note that "stream" is the most efficient implementation. "bytes" makes a copy and "string" makes two copies.

Reply:
From our discussion, we can get to memory footprint optimizations in the future.